### ISSUE TITLE
Accuracy drops from very high to 0 after very few steps on T5 encoder model

### ISSUE THREAD

**Issue by sebastian-nehrdich** [UKPLab/sentence-transformers#2382]
I am trying to train a model with MultipleNegativesRankingLoss on a bilingual dataset with 150k sentences. The training works well if I use LaBSE as a base model and it reaches very high precision after a few 1000 steps.
However, when I initialize the same training process with the encoder stack of a T5 model, the precision is first very good, but after some steps it drops to zero, i.e. the model totally unlearns whatever it could do before.
When I run the model in eval mode, the precision is great. However, after training it on the dataset for just a few steps, the weights are ruined. 
This is the encoder: https://huggingface.co/buddhist-nlp/madlad-encoder

These are the training parameters:

`student_model.fit(train_objectives=[(train_dataloader, train_loss)],
          evaluator=evaluation.SequentialEvaluator(evaluators, main_score_function=lambda scores: np.mean(scores)),
          epochs=num_epochs,
          warmup_steps=num_warmup_steps,
          evaluation_steps=num_evaluation_steps,
          output_path=output_path,
          save_best_model=True,
          use_amp=True,
          optimizer_params= {'lr': 2e-7, 'eps': 1e-6}
          )

`
The train loss is defined as such: 
`train_loss = losses.MultipleNegativesRankingLoss(model=student_model)
`
I am loading the model simply with SentenceTransformer(model_path). It doesn't matter if I train in half precision or full precision, the training fails in both cases. If I switch back to LaBSE, the script trains succesfully.
